packages/data/src/components/use-select/index.js

```diff
@@ -227,7 +230,7 @@ export default function useSelect( _mapSelect, deps ) {
 			unsubscribers.forEach( ( unsubscribe ) => unsubscribe?.() );
 			renderQueue.flush( queueContext );
 		};
-	}, [ registry, trapSelect, depsChangedFlag, isWithoutMapping ] );
+	}, [ registry, trapSelect, hasMappingFunction, ...( deps || [] ) ] );
```

Comment:
```suggestion
	}, [ registry, trapSelect, hasMappingFunction, ...( deps ?? [] ) ] );
```

Reply:
I believe spreading in the dependencies array disobey the [lint rule](https://github.com/facebook/react/issues/14920), which we haven't enabled but eventually I'd want to.

Reply:
Hm. I don't know if we really want to obey that lint rule here because I think the problem is treating the deps array as an array instead of as individual deps. Did you have an alternative suggestion @kevin940726? Or was the `useMemo` workaround a way of accomplishing the same behavior but avoiding an `ignore` rule for the linter?

Reply:
@ZebulanStanphill I wondered about `||` vs. `??` but wasn't sure. The main purpose of the fallback is to make sure we don't try and spread a non-spreadable value, so I used `||` since it catches other falsey values.

I guess in any situation where we get a non-spreadable value that isn't `undefined` or `null` that's also a failure anyway, so it would be okay I guess to let it crash if we use `??`, but maybe also fine to avoid the crash with `||` if it's not necessary or important?

Any thoughts?

Reply:
@dmsnell (didn't see your latest comment while writing the prior one), I generally prefer using `??` over `||` wherever possible since it's more explicit and hides fewer bugs.

As you stated, `deps` should always be either an array or nullish. If it's something like `''` or `false`, then something is very wrong, and it would be better to throw.

Reply:
Nah, I agree with @ZebulanStanphill that adding a clear comment above it for this specific special case should be enough.

Or else, we could use some hack with multiple `useRef`s to micmic the same behavior. It should work, but doesn't worth it at the time.

Reply:
thanks for the review both of you!

> it's fine to break a linter rule as long as you add a comment clearly explaining why this case is special.
> adding a clear comment above it

no existing comment because there's no linter to ignore 🙃 

> Or else, we could use some hack with multiple useRefs to micmic the same behavior. It should work, but doesn't worth it at the time.

Yeah I feel like this is one of those things where we really want to avoid making awkward code choices to avoid exempting ourselves from a lint rule. This is a special case whereby we are passing in `deps` from the caller, not at all like what it appeared the lint rule is meant to cover, unless I'm misunderstanding it.

> I generally prefer using ?? over || wherever possible since it's more explicit and hides fewer bugs.

I would normally agree 100% but in this case it's more of a question of which behavior we want rather than what is explicit. Still, I like your leaning on crashing, though it's a little inconsistent since we have the try/catch around `mapSelect`. I'll switch to `??` + crashing for now.

Reply:
> Yeah I feel like this is one of those things where we really want to avoid making awkward code choices to avoid exempting ourselves from a lint rule. This is a special case whereby we are passing in `deps` from the caller, not at all like what it appeared the lint rule is meant to cover, unless I'm misunderstanding it.

Yeah, this special case is what they [discourage](https://www.npmjs.com/package/eslint-plugin-react-hooks#advanced-configuration) in the first place though. Personally I'd go for a hacky solution that obey the lint rule just that we can be sure we will not accidentally ship bugs in future refactors. Most of the hacky solutions could be hidden under custom hooks anyway. I don't mind disabling the rule here simply because it's already breaking the rule way before this change 😆 .

Reply:
thanks for the review both of you!

> it's fine to break a linter rule as long as you add a comment clearly explaining why this case is special.
> adding a clear comment above it

no existing comment because there's no linter to ignore 🙃 

> Or else, we could use some hack with multiple useRefs to micmic the same behavior. It should work, but doesn't worth it at the time.

Yeah I feel like this is one of those things where we really want to avoid making awkward code choices to avoid exempting ourselves from a lint rule. This is a special case whereby we are passing in `deps` from the caller, not at all like what it appeared the lint rule is meant to cover, unless I'm misunderstanding it.

> I generally prefer using ?? over || wherever possible since it's more explicit and hides fewer bugs.

I would normally agree 100% but in this case it's more of a question of which behavior we want rather than what is explicit so I'm way down at around 99% agreement 🙃. Still, I like your leaning towards crashing, though it's a little inconsistent since we have the try/catch around `mapSelect`. I'll switch to `??` + crashing for now.